packages/typir/src/utils/utils-type-comparison.ts

```diff
@@ -80,166 +81,173 @@ export interface IndexedTypeConflict extends TypirProblem {
     // 'undefined' means type or information is missing, 'string' is for data which are no Types
     expected: Type | undefined; // first, left
     actual: Type | undefined; // second, right
-    index: number | string;
+    // index OR name should be specified
+    propertyIndex?: number;
+    propertyName?: string;
     subProblems: TypirProblem[];
 }
 export const IndexedTypeConflict = 'IndexedTypeConflict';
 export function isIndexedTypeConflict(problem: unknown): problem is IndexedTypeConflict {
     return isConcreteTypirProblem(problem, IndexedTypeConflict);
 }
 
-export function checkNameTypePairs(left: NameTypePair[], right: NameTypePair[], checkNames: boolean, relationToCheck: (l: Type, r: Type) => (TypirProblem | undefined)): IndexedTypeConflict[] {
+export type TypeToCheck = Type | NameTypePair | undefined | InferenceProblem[];
```

Comment:
I am unsure what the "NameTypePair" is good for. I am going to check back with you and/or Markus and potentially will add one or two comments afterwards.

Reply:
I talked to Markus and had another visit to the code. This is clear now and resolved for me.